pkg/agent/controller/traceflow/packetin.go

```diff
@@ -176,8 +178,17 @@ func (c *Controller) parsePacketIn(pktIn *ofctrl.PacketIn) (*opsv1alpha1.Tracefl
 				return nil, nil, err
 			}
 		}
-		if tunnelDstIP != "" && tunnelDstIP != c.nodeConfig.NodeIPAddr.IP.String() {
-			ob.TunnelDstIP = tunnelDstIP
+		var outputPort uint32
+		if match = getMatchRegField(matchers, uint32(openflow.PortCacheReg)); match != nil {
+			outputPort, err = getInfoInReg(match, nil)
+			if err != nil {
+				return nil, nil, err
+			}
+		}
+		if outputPort == config.DefaultTunOFPort || outputPort == config.HostGatewayOFPort {
+			if tunnelDstIP != "" {
+				ob.TunnelDstIP = tunnelDstIP
+			}
```

Comment:
could you add a comment here, I'm struggling to understand what's happening?

is `ob.tunnelDstIP` set to anything in noEncap mode?

Reply:
Added comments here.